### PR TITLE
[depends] Bump openssl and gnutls

### PR DESCRIPTION
**Pull request by wsnipex**
## Description

bump openssl and gnutls to latest stable

## Motivation and context

openssl 1.1.0 is deprecated, out of support and has know security issues.
This reportedly also fixes an issue with python requests and some URLs on android, see https://forum.kodi.tv/showthread.php?tid=363015&pid=3043991#pid3043991
Given that, this should also warrant a backport to 19.2

## Types of change

- [X] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by garbear** [comment]
OpenSSL is also bumped in https://github.com/xbmc/xbmc/pull/19432, FWIW. Looks like there were other changes and I think mariadb needed modification.

**Comment by fuzzard** [comment]
Yeah, as mentioned i came across the pthread_atfork issue in that other PR. my simplistic fix was the following

https://github.com/xbmc/xbmc/pull/19432/commits/437f9e1b4e6ab6c1d88b5fe5978a80ac867429d6

I had a fair bit of other changes. Most was refactoring to use openssl platforms for their respected platform instead of just using linux for android and linux. I introduced a conf for our apple platforms to support ios/tvos/osx (apple silicon and intel). Was just easier than having to do sed replaces on assumptions the openssl team do about some of those platforms that arent relevant to us.

To top that all off, most of our existing sed replaces arent actually valid anymore.

I wasnt really going to comment on this, as the apple specific stuff can always be done later, and the linux/android stuff was more of cleanup thats probably not really required. So fix this as you need, and i'll deal with the apple stuff later

**Comment by wsnipex** [comment]
thanks!

**Comment by wsnipex** [comment]
hmm, that didn't work, let's try something else

**Review by fuzzard — APPROVED** [review]
Always hate having to carry a patch, but if it works, all good.

File: tools/depends/target/mariadb/04-pthread.patch [hunk 1]
```diff
@@ -0,0 +1,11 @@
+--- a/CMakeLists.txt	2021-06-28 08:00:02.761728364 +0000
++++ b/CMakeLists.txt	2021-06-28 08:00:29.198289581 +0000
+@@ -293,7 +293,7 @@
+     IF(OPENSSL_FOUND)
+       ADD_DEFINITIONS(-DHAVE_OPENSSL -DHAVE_TLS)
+       SET(SSL_SOURCES "${CC_SOURCE_DIR}/libmariadb/secure/openssl.c")
+-      SET(SSL_LIBRARIES ${OPENSSL_SSL_LIBRARY} ${OPENSSL_CRYPTO_LIBRARY})
++      SET(SSL_LIBRARIES ${OPENSSL_SSL_LIBRARY} ${OPENSSL_CRYPTO_LIBRARY} ${_OPENSSL_LDFLAGS_OTHER})
```
Comment (garbear) [inline tools/depends/target/mariadb/04-pthread.patch:8]: Is the leading `_` on the variable `${_OPENSSL_LDFLAGS_OTHER}` wanted? I haven't seen a leading underscore used for CMake variables often

Comment (wsnipex) [reply tools/depends/target/mariadb/04-pthread.patch:8]: yes

**Pull request merged**